src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -49,6 +49,14 @@ private[actions] abstract class RefreshActionBase(
 
   protected lazy val previousIndexLogEntry = previousLogEntry.asInstanceOf[IndexLogEntry]
 
+  override protected final def getNumBucketsConfig(spark: SparkSession): Int = {
```

Comment:
Let's add `// Refresh maintains the same number of buckets as the existing index because ...` and we can mention the about the full refresh here.